manticore/utils/config.py

```diff
@@ -0,0 +1,208 @@
+        """
+        if name not in self._vars:
+            raise ConfigError("f{self.name}.{name} not defined.")
+
+        return self._vars[name].description
+
+    def was_set(self, name: str) -> bool:
+        return self._vars[name].was_set
+
+    def _var_object(self, name: str) -> _var:
+        return self._vars[name]
+
+    def __getattr__(self, name):
+        if name not in self._vars:
+            raise AttributeError(f"Group '{self.name}' has no variable '{name}'")
+        return self._vars[name].value
+
+    def __setattr__(self, name, new_value):
+        self._vars[name].value = new_value
+
+    def __iter__(self):
+        return iter(self._vars)
+
+    def __contains__(self, key):
+        return key in self._vars
+
+
+def get_group(name: str):
+    """
+    Get a configuration variable group named |name|
+    """
+    global _groups
+
+    if name in _groups:
+        return _groups[name]
+
+    group = _group(name)
+    _groups[name] = group
+
+    return group
+
+
+def save(f):
+    """
+    Save current config state to an ini file stream identified by |f|
+
+    :param f: where to write the config file
+    """
+    global _groups
+
+    c = configparser.ConfigParser()
+    for group_name, group in _groups.items():
+        if not any(group.was_set(v) for v in group):
+            continue
+        c.add_section(group_name)
+        for var in group:
+            if not group.was_set(var):
+                continue
+            c.set(group_name, var, str(getattr(group, var)))
+    c.write(f)
+
+
+def parse_ini(f):
+    """
+    Load an ini-formatted configuration from file stream |f|
+
+    :param file f: Where to read the config.
+    """
+
+    # This currently does some hacky ast parsing on the literals, but this is in service
+    # of having a simpler, ini-style configuration without external dependencies, like
+    # a YAML parser, and ini files do not have typed values. 
+
+    c = configparser.ConfigParser()
+    c.read_file(f)
+    for section_name in c.sections():
+        group = get_group(section_name)
+
+        for key, v in c.items(section_name):
+            try:
+                val = ast.literal_eval(v)
+            except ValueError:
+                val = v
+
+            group.update(key)
+            setattr(group, key, val)
+
+
+def load_overrides(path=None):
+    """
+    Load config overrides from the ini file at |path|, or from default paths. If a path
+    is provided and it does not exist, raise an exception
+
+    Default paths: ./mcore.ini, ./.mcore.ini, ./manticore.ini, ./.manticore.ini.
+    """
+    possible_names = ['mcore.ini', 'manticore.ini']
+    names = [os.path.join('.', ''.join(x)) for x in product(['', '.'], possible_names)]
+
+    if path is not None:
+        names = [path]
+
+    for name in names:
+        if os.path.exists(name):
+            logger.info(f'Reading configuration from {name}')
+            with open(name, 'r') as ini_f:
```

Comment:
Funfact, there is a [TOCTOU](https://en.wikipedia.org/wiki/Time_of_check_to_time_of_use)/race condition here: we check if file exists and later open it.

An [EAPF style](https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html) is less bug-prone in that matter (as long as the exception is caught early on, which is not always obvious)